content/auth/basic.textile

```diff
@@ -0,0 +1,112 @@
+
+The following is an example of using basic authentication:
+
+REALTIME-CODE 
+
+```[jsall]
+  var realtime = new Ably.Realtime({ key: '{{API_KEY}}' });
+```
+
+```[ruby]
+  realtime = Ably::Realtime.new(key: '{{API_KEY}}')
+```
+
+```[java]
+  ClientOptions options = new ClientOptions();
+  options.key = "{{API_KEY}}";
+  AblyRealtime realtime = new AblyRealtime(options);
+```
+
+```[swift]
+  let realtime = ARTRealtime(key: "{{API_KEY}}")
+```
+
+```[objc]
+  ARTRealtime *realtime = [[ARTRealtime alloc] initWithKey:@"{{API_KEY}}"];
+```
+
+```[csharp]
+  AblyRealtime realtime = AblyRealtime("{{API_KEY}}");
+```
+
+REST-CODE 
+
+```[jsall](code-editor:realtime/connection)
+  var rest = new Ably.Rest({ key: '{{API_KEY}}' });
+```
+
+```[ruby]
+  rest = Ably::Rest.new(key: '{{API_KEY}}')
+```
+
+```[python]
+  rest = AblyRest(key='{{API_KEY}}')
+```
+
+```[php]
+  $rest = new Ably\AblyRest(array('key' => '{{API_KEY}}'));
+```
+
+```[java]
+  ClientOptions options = new ClientOptions();
+  options.key = "{{API_KEY}}";
+  AblyRest rest = new AblyRest(options);
+```
+
+```[csharp]
+  AblyRest rest = new AblyRest("{{API_KEY}}");
+```
+
+```[swift]
+  let rest = ARTRest(key: "{{API_KEY}}")
+```
+
+```[objc]
+  ARTRest *rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+```
+
+```[go]
+  client, err := ably.NewRestClient(ably.ClientOptions("{{API_KEY}}"))
+```
+
+h2(#diagram). Basic auth architecture
+
+The process used by Ably SDKs to authenticate with Ably using basic authentication is illustrated in the following diagram:
+
+<a href="/images/diagrams/Ably-API-Auth1.gif" target="_blank">
+  <img src="/images/diagrams/Ably-API-Auth1.gif" style="width: 100%" alt="Basic authentication process diagram">
+</a>
+
+h2(#when). When to use basic auth
+
+Ably recommends that basic authentication is only used server-side because of the following potential issues:
+
+* The secret is passed directly by the client to Ably, so it is only permitted for connections that are over TLS, to prevent the key secret being intercepted.
+* All of the configured "capabilities":/auth/capabilities of the key are implicitly possible in any request, and clients that legitimately obtain this key may then abuse the rights for that key.
+* A client that authenticates using an API key can claim any client ID it chooses. Therefore this client ID cannot be trusted to represent the genuine identity of the client. Client IDs should be assigned by the server, once the client's credentials have been authenticated.
+
+<!-- CAVEAT-INPUT-->
+
+When selecting an Ably SDK for implementing basic authentication with Ably, you don't need to use the realtime interface. 
+
+As basic authentication is primarily designed for authenticating a secure server, in this situation, and assuming the server is only used for authentication, it is more efficient to use the REST interface, as the overhead associated with maintaining a realtime connection, is not required.
```

Comment:
"associated with maintaining a realtime connection, is not required."

Remove comma at the end of the sentence. Might be worth a bit of a rewrite of the entire sentence as it's on the edge of being confusing with all the commas.

Reply:
Good point - reworded the whole paragraph.